apps/designer/app/designer/shared/image-manager/image.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from "react";
+import { styled } from "@webstudio-is/design-system";
+import { Image as WebstudioImage, loaders } from "@webstudio-is/image";
+import env from "~/shared/env";
+import type { RenderableAsset } from "../assets";
+import brokenImage from "~/shared/images/broken-image-placeholder.svg";
+
+type ImageProps = {
+  renderableAsset: RenderableAsset;
+  alt: string;
+  width: number;
+};
+
+const StyledWebstudioImage = styled(WebstudioImage, {
+  position: "absolute",
+  width: "100%",
+  height: "100%",
+  objectFit: "contain",
+
+  // This is shown only if image was not loaded and broken
+  // from spec:
+  // The pseudo-elements generated by ::before and ::after are contained by the element's formatting box,
+  // and thus don't apply to replaced elements such as <img>, or to <br> elements
+  // But broken image is replaced element so this style is applied
+  "&::after": {
+    content: "' '",
+    position: "absolute",
+    width: "100%",
+    height: "100%",
+    left: 0,
+    top: 0,
+    backgroundSize: "contain",
+    backgroundRepeat: "no-repeat",
+    backgroundPosition: "center",
+    backgroundImage: `url(${brokenImage})`,
+  },
+});
+
+export const Image = ({ renderableAsset, alt, width }: ImageProps) => {
+  const optimize = renderableAsset.status === "uploaded";
+  const asset =
+    renderableAsset.status === "uploaded"
+      ? renderableAsset.asset
+      : renderableAsset.preview;
+  const remoteLocation =
+    renderableAsset.status === "uploaded" &&
+    renderableAsset.asset.location === "REMOTE";
+
+  // Avoid image flickering on switching from preview to asset (during upload)
+  const decoding = renderableAsset.preview === undefined ? "async" : "sync";
+
+  const loader = useMemo(() => {
+    if (remoteLocation) {
+      return loaders.cloudflareImageLoader({
+        resizeOrigin: env.RESIZE_ORIGIN,
+      });
+    }
+
+    return loaders.localImageLoader();
+  }, [remoteLocation]);
```

Comment:
sounds like you could make useLoader hook outside of the component

Reply:
Locally still possible that some images are remote and are not

Reply:
Just better logic organization. I always try to pull all the logic out of component into a hook next to it.
const useLoader = (assetContainer) => {
  ...
}